AppCenterDataStorage/AppCenterDataStorage/MSDataStore.m

```diff
@@ -214,25 +214,27 @@ - (void)replaceWithPartition:(NSString *)partition
                    completionHandler:completionHandler];
 }
 
-- (NSError *)generateDisabledError:(NSString *)operation documentId:(NSString *)documentId {
-  NSError *error = [[NSError alloc] initWithDomain:kMSACErrorDomain
-                                              code:MSACDisabledErrorCode
-                                          userInfo:@{NSLocalizedDescriptionKey : kMSACDisabledErrorDesc}];
-  MSLogError([MSDataStore logTag], @"Not able to %@ the document ID: %@; error: %@", operation, documentId, [error localizedDescription]);
-  return error;
-}
-
 - (void)readWithPartition:(NSString *)partition
                documentId:(NSString *)documentId
              documentType:(Class)documentType
               readOptions:(MSReadOptions *_Nullable)__unused readOptions
         completionHandler:(MSDocumentWrapperCompletionHandler)completionHandler {
   @synchronized(self) {
+
+    // Check preconditions.
+    NSError *error;
     if (![self canBeUsed] || ![self isEnabled]) {
-      NSError *error = [self generateDisabledError:@"read" documentId:documentId];
+      error = [self generateDisabledError:@"read" documentId:documentId];
+    }
+    if (!error && ![MSDocumentUtils isSerializableDocument:documentType]) {
```

Comment:
Can we improve the readability of two sequential `if` checks into an `if...else if...` check:
```
if (![self canBeUsed] || ![self isEnabled]) {...}
else if ![MSDocumentUtils isSerializableDocument:documentType]) {...}
```
Assuming `[self generateDisabledError:@"read" documentId:documentId];` always returns non nil.